src/main/java/com/google/api/generator/engine/ast/MethodDefinition.java

```diff
@@ -139,6 +139,9 @@ public MethodDefinition build() {
 
       MethodDefinition method = autoBuild();
 
+      Preconditions.checkState(
+          !method.returnType().equals(TypeNode.NULL), "Method returnType can not be null");
```

Comment:
Nit: "return type"

Reply:
Also, consider rephrasing this as per the comment below. For example, readers might wonder why they cannot return null, because `return null` is valid Java. However, what we mean to say is something like, "Null is not a valid method return type."

Reply:
agree. Thanks.